pilot/pkg/serviceregistry/kube/controller/pod.go

```diff
@@ -147,6 +148,25 @@ func (pc *PodCache) getPodByIP(addr string) *v1.Pod {
 	return item.(*v1.Pod)
 }
 
+// getPod first checks if a pod with ip exists in the cache. If it exists, it just returns
+// from the cache otheriwse it loads the pod from k8s.
+func (pc *PodCache) getPod(addr string, namespace string) *v1.Pod {
+	pod := pc.getPodByIP(addr)
+	if pod != nil {
+		return pod
+	}
+	pods, err := pc.c.client.CoreV1().Pods(namespace).List(metav1.ListOptions{})
```

Comment:
We can get the pod name from the endpoints targetRef

Reply:
That is a good point. Let me change like that.